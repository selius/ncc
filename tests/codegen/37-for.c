int main()
{
	int i;
	int a;

	for (i = 0; i < 5; i++) {
		__print_int(i);
	}

	a = 0;
	for (i = 1; i <= 1000; i++) {
		a += i;
	}
	__print_int(a);

	for (; 0;) {
		__print_int(999);
		break;
	}
	

	i = 0;
	for (;;) {
		if (i > 100) {
			break;
		}
		++i;
	}
	__print_int(i);

	a = 0;
	for (i = 0; i < 10; i++) {
		a += i;
		continue;
		a *= i;
	}
	__print_int(a);
	
	return 0;
}
