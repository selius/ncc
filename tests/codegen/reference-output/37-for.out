0
1
2
3
4
500500
101
45
